JWBestPracticeApps/Google IMA Ads/Google IMA Ads/ViewController.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ViewController.swift
+//  Google IMA Ads
+//
+//  Created by David Almaguer on 09/08/21.
+//
+
+import UIKit
+import JWPlayerKit
+
+class ViewController: JWPlayerViewController {
+
+    private let vmapUrlString = "https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/ad_rule_samples&ciu_szs=300x250&ad_rule=1&impl=s&gdfp_req=1&env=vp&output=vmap&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ar%3Dpremidpost&cmsid=496&vid=short_onecue&correlator="
+    private let videoUrlString = "https://cdn.jwplayer.com/videos/CXz339Xh-sJF8m8CA.mp4"
+    private let posterUrlString = "https://cdn.jwplayer.com/thumbs/CXz339Xh-720.jpg"
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Set up the player.
+        setUpPlayer()
+    }
+
+    /**
+     Sets up the player with an advertising configuration.
+     */
+    private func setUpPlayer() {
+        let videoUrl = URL(string:videoUrlString)!
+        let posterUrl = URL(string:posterUrlString)!
+        let vmapURL = URL(string: vmapUrlString)!
+
+        do {
+            // First, use the JWPlayerItemBuilder to create a JWPlayerItem that will be used by the player configuration.
+            let playerItembuilder = JWPlayerItemBuilder()
+                .file(videoUrl)
+                .posterImage(posterUrl)
+
+            // Build the player item. This method can throw so be sure to handle the error.
+            let playerItem = try playerItembuilder.build()
+
+            // Second, use the JWImaAdvertisingConfigBuilder to create a JWAdvertisingConfig that will be used by the player configuration.
+            let adConfigBuilder = JWImaAdvertisingConfigBuilder()
+                // Set the VMAP url for the builder to use.
+                .vmapURL(vmapURL)
+
+            // Build the IMA advertising configuration. This method can throw so be sure to handle the error.
+            let adConfig = try adConfigBuilder.build()
```

Comment:
To simplify, let's chain .build() after .vmapURL() to make the code more concise. This also will show the developer why we did the builder pattern. Separating it out makes it look bulkier.

Reply:
Addressed